### PR TITLE
[3055] prevent from passing different cx and getValue functions to children components when the results of those functions remain the same as previous render.

### PR DESCRIPTION
**Pull request by manvydasu**
During the render of Select component, functions cx and getValue are recreated as new functions everytime and are passed to children components. Sometimes the result of cx and getValue is same as previous render, but the functions are still re-created thus causing unnecessary re-renders as result. The issue was originally described here: https://github.com/JedWatson/react-select/issues/3055

Would love if someone could suggest a cleaner way to fix this.

### CODE REVIEW

**Comment by Rall3n** [comment]
I think a better/cleaner solution for both would be to create the functions on class level and just add them to the `commonProps` object.

```jsx
getValue = () => this.state.selectValue;

cx = (...args) => classNames(this.props.classNamePrefix, ...args);

/* ... */
getCommonProps() {
    const { /* ... */ cx, getValue, props } = this;

    /* ... */

    return {
        cx,
        getValue,
        /* ... */
    };
}
```

No need for memoization and no need for "creation"-functions.

**Comment by manvydasu** [comment]
@Rall3n 
Yeah, you're right. I over engineered that originally. Updated just as you suggested.

**Comment by bladey** [comment]
Look good to me, @JedWatson to review.

File: packages/react-select/src/Select.js [hunk 1]
```diff
@@ -742,14 +742,15 @@ export default class Select extends Component<Props, State> {
     };
   }
 
+  getValue = () => this.state.selectValue;
+
+  cx = (...args: any) => classNames(this.props.classNamePrefix, ...args);
```
Comment (Methuselah96) [inline packages/react-select/src/Select.js:747]:
```suggestion
  cx = (...args: Array<any>) => classNames(this.props.classNamePrefix, ...args);
```

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

**Review by JedWatson — APPROVED** [review]

**Comment by JedWatson** [comment]
Thanks @manvydasu

**Pull request merged**